### PR TITLE
Update RTL5 for 0.9

### PR DESCRIPTION
**Pull request by ricardopereira**

### CODE REVIEW

**Comment by mattheworiordan** [comment]
Thanks @tcard, I concur

**Comment by ricardopereira** [comment]
@mattheworiordan @tcard PTAL

File: Spec/RealtimeClientChannel.swift [hunk 1]
```diff
@@ -1079,7 +1079,7 @@ class RealtimeClientChannel: QuickSpec {
 
             describe("detach") {
                 // RTL5a
-                it("if state is INITIALISED, DETACHED or DETACHING nothing is done") {
+                it("if state is INITIALISED and DETACHED nothing is done") {
```
Comment (mattheworiordan) [inline Spec/RealtimeClientChannel.swift]: What do you mean INITIALISED and DETACHED, it can't be both.  Also, `INITIALIZED` spelling please

File: Spec/RealtimeClientChannel.swift [hunk 2]
```diff
@@ -1118,6 +1111,76 @@ class RealtimeClientChannel: QuickSpec {
...
+                // RTL5i
+                it("if the channel is in a pending state ATTACHING, do the detach operation after the completion of the pending request") {
+                    let options = AblyTests.commonAppSetup()
+                    let client = ARTRealtime(options: options)
+                    defer { client.dispose(); client.close() }
+                    let channel = client.channels.get("foo")
+
+                    waitUntil(timeout: testTimeout) { done in
+                        let partialDone = AblyTests.splitDone(2, done: done)
+                        channel.once(.Attaching) { stateChange in
```
Comment (mattheworiordan) [inline Spec/RealtimeClientChannel.swift]: Surely you need to check that the state moves to the `ATTACHED` state i.e. "do the detach operation after the completion of the pending request".  The pending operation is to attach the channel

File: Spec/RealtimeClientChannel.swift [hunk 3]
```diff
@@ -1118,6 +1111,76 @@ class RealtimeClientChannel: QuickSpec {
...
+                            expect(stateChange.reason).to(beNil())
+                            expect(stateChange.current).to(equal(ARTRealtimeChannelState.Detaching))
+                            expect(stateChange.previous).to(equal(ARTRealtimeChannelState.Attached))
+                            channel.detach()
+                            done()
+                        }
+                        channel.detach()
+                    }
+
+                    expect(detachedCount) == 1
```
Comment (mattheworiordan) [inline Spec/RealtimeClientChannel.swift]: This is racy, you don't allow enough time for the channel to potentially move into other states as you complete the test as soon as it reaches the detaching state which is triggered by the `channel.detach()`.  I would at a minimum add a timer to avoid this race and also check how many DETACHING states you see.

File: Spec/RealtimeClientChannel.swift [hunk 4]
```diff
@@ -1376,6 +1439,32 @@ class RealtimeClientChannel: QuickSpec {
...
+                            expect(error).to(beNil())
+                            done()
+                        }
+                    }
+
+                    channel.setSuspended(ARTStatus.state(.Ok))
+                    expect(channel.state).to(equal(ARTRealtimeChannelState.Suspended))
+
+                    waitUntil(timeout: testTimeout) { done in
+                        channel.detach() { error in
```
Comment (mattheworiordan) [inline Spec/RealtimeClientChannel.swift]: Should you not check that the previous state is `SUSPENDED` and not `DETACHING` as it states " the detach request transitions the channel immediately to the DETACHED state"

File: Spec/RealtimeClientChannel.swift (hunk 1 above)
Comment (ricardopereira) [inline Spec/RealtimeClientChannel.swift]: Done d771c91.

File: Spec/RealtimeClientChannel.swift (hunk 3 above)
Comment (ricardopereira) [inline Spec/RealtimeClientChannel.swift]: Done f13ccd0.

File: Spec/RealtimeClientChannel.swift (hunk 2 above)
Comment (ricardopereira) [inline Spec/RealtimeClientChannel.swift]: Done efd6f48.

File: Spec/RealtimeClientChannel.swift (hunk 4 above)
Comment (ricardopereira) [inline Spec/RealtimeClientChannel.swift]: Done 025ea74.

**Comment by mattheworiordan** [comment]
Looking good...

**Comment by ricardopereira** [comment]
but I did a regression :/ I'm looking at it right now.

File: Spec/RealtimeClientPresence.swift [hunk 5]
```diff
@@ -226,7 +226,7 @@ class RealtimeClientPresence: QuickSpec {
             }
 
             // RTP5
-            context("Channel state change side effects") {
+            pending("Channel state change side effects") {
```
Comment (ricardopereira) [inline Spec/RealtimeClientPresence.swift]: I deal with this on a later PR.
